### ISSUE TITLE
Default segmentation image augmentation, why pad 0 to the mask

### ISSUE THREAD

**Issue by weiaicunzai** [dmlc/gluon-cv#1111]
Hi, guys, I've noticed the default gluon-cv segmentation augmentation also pad 0 to the mask when resizing, why use 0, would this affect the performance of the network predicting the index 0 class? Or causing the training sample imbalance problem(since there might be too many 0 class pixels)?

in https://github.com/dmlc/gluon-cv/blob/49056d4892382fca067216d6271d1633d869cb8d/gluoncv/data/segbase.py
```python
img = ImageOps.expand(img, border=(0, 0, padw, padh), fill=0)
mask = ImageOps.expand(mask, border=(0, 0, padw, padh), fill=0) # why fill 0??
```

Thanks in advance.

**Comment by bryanyzhu** [comment]
We have a specific [routine](https://github.com/dmlc/gluon-cv/blob/master/gluoncv/data/cityscapes.py#L40) to transform the GT mask to class 1~N, and the ignore_class will always be class 0. So we can safely pad 0 to the mask for training the models.

Take Cityscapes as an example, there are 19 classes. We first transform the GT mask to class 1 to 19, and all other ignored pixels to class 0. Then we minus 1 to get the target label, which is zero based, from 0 to 18. So you can use standard cross-entropy to compute the loss.

**Comment by weiaicunzai** [comment]
> We have a specific [routine](https://github.com/dmlc/gluon-cv/blob/master/gluoncv/data/cityscapes.py#L40) to transform the GT mask to class 1~N, and the ignore_class will always be class 0. So we can safely pad 0 to the mask for training the models.
> 
> Take Cityscapes as an example, there are 19 classes. We first transform the GT mask to class 1 to 19, and all other ignored pixels to class 0. Then we minus 1 to get the target label, which is zero based, from 0 to 18. So you can use standard cross-entropy to compute the loss.

Ok, thank you for clarifying, guess I missed the 0 is the ignored class index part.
